### PR TITLE
Fixed review list ajax if product not exist redirect to 404 page #13102

### PR DESCRIPTION
**Pull request by ananth-iyer**
### Description
Issue was review list ajax call giving php error log or error number if invalid product id. It should be 404 page.

### Fixed Issues (if relevant)
1. magento/magento2#13102, Page was not redirect to 404 page.

### Manual testing scenarios
1. Use invalid product id in review list ajax
2. For example: `<Magento store url>/review/product/listAjax/id/{{non existent id}/`

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=15369) <br/>All committers have signed the CLA.

**Comment by larsroettig** [comment]
Hi @Ananth747, can you sign pls our contributor License Agreement.
I start now with review and testing.

best regards,

Lars

File: app/code/Magento/Review/Controller/Product/ListAjax.php [hunk 1]
```diff
@@ -14,12 +14,15 @@ class ListAjax extends ProductController
...
      */
     public function execute()
     {
         if (!$this->initProduct()) {
-            throw new LocalizedException(__('Cannot initialize product'));
+            /** @var \Magento\Framework\Controller\Result\Forward $resultForward */
+            $resultForward = $this->resultFactory->create(ResultFactory::TYPE_FORWARD);
+            $resultForward->forward('noroute');
+            return $resultForward;
         } else {
```
Comment (larsroettig) [inline app/code/Magento/Review/Controller/Product/ListAjax.php]: It is unnecessary else statement you can remove this. 

```php
 /**
     * Show list of product's reviews
     *
     * @return \Magento\Framework\App\ResponseInterface|\Magento\Framework\Controller\ResultInterface
     */
    public function execute()
    {
        if (!$this->initProduct()) {
            /** @var \Magento\Framework\Controller\Result\Forward $resultForward */
            $resultForward = $this->resultFactory->create(ResultFactory::TYPE_FORWARD);
            return $resultForward->forward('noroute');
        }

        return $this->resultFactory->create(ResultFactory::TYPE_LAYOUT);
    }
```

**Review by larsroettig — CHANGES_REQUESTED** [review]

**Review by larsroettig — APPROVED** [review]

**Comment by ananth-iyer** [comment]
@larsroettig @magento-cicd2 I've signed CLA when it will be accepted?

**Comment by sidolov** [comment]
Hi @Ananth747 , looks like your commits were pushed with the different email than on GitHub profile, please, add email from commits to your account and issue with CLA will be fixed.

**Comment by ananth-iyer** [comment]
@magento-cicd2 @sidolov @larsroettig 
I've added the email that was used for commit. Now CLA check is passed.
Sorry for the misunderstanding.

**Comment by magento-engcom-team** [comment]
Hi @Ananth747. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.7. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Pull request merged**